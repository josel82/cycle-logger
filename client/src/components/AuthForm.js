import React, { Component } from 'react';

class AuthForm extends Component {

    constructor(props){
        super(props);
        this.state = {
            email: '',
            password: '',
            emailInvalid: false,
            passwordInvalid: false
        }
    }
    onEmailChange = (email) => {
        this.setState({email});
        if(email.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i)){
            this.setState({emailInvalid:false});
        }else{
            this.setState({emailInvalid:true});
        }
    }
    onPasswordChange = (password) => {        
        this.setState({password});
        if(password.match(/^.{8,32}$/)){
            this.setState({passwordInvalid:false});
        }else{
            this.setState({passwordInvalid:true});
        }
    }
    handleOnSubmit = (e) => {
        e.preventDefault();
    
        this.props.handleAuthorization(this.state.email, this.state.password);
    }

    render(){
        return (
            <div className="form-container">
                <h1 className="form-title">{this.props.action}</h1>
                <div className="form-box">
                    <form onSubmit={this.handleOnSubmit} className="form form-auth">
                        <input 
                            type="text" 
                            className="form__input" 
                            name="email" 
                            value={this.state.email}
                            onChange={this.onEmailChange}
                             />
                        <input 
                            type="password" 
                            className="form__input" 
                            name="password" 
                            value={this.state.password}
                            onChange={this.onPasswordChange} 
                            />
                        <button 
                            className="btn btn-primary btn-md"
                            disabled={this.state.emailInvalid || this.state.passwordInvalid}
                            >
                            {this.props.action}
                        </button>
                    </form>
                </div>
            </div>  
        );
    }
}

export default AuthForm;