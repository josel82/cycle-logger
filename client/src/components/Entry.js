import React, { Component } from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import { withRouter } from 'react-router-dom';
import numeral from 'numeral';
import '../formats/mg';


export class Entry extends Component {
    constructor(props){
      super(props);
      this.date = moment(this.props.timestamp).format('Do MMM, YYYY');
      this.formatedQty = numeral(this.props.quantity).format('0[.]00 mg');
    }
    
    redirectToEditEntryPage = () => {
      this.props.history.push(`/dashboard/edit/${this.props.id}`);
    }
    render(){
      return (
        <tr onClick={this.redirectToEditEntryPage}>
              <td><p>{this.props.compound}</p></td>
              <td><p>{this.formatedQty}</p></td>
              <td>
                <p>{this.date}</p>
              </td>
            </tr>
          )
        } 
  };

  const EntryWithRouter = withRouter(Entry)

  export default connect()(EntryWithRouter);