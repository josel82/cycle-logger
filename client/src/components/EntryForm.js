import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { faTrashAlt } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

class EntryForm extends Component {
    constructor(props){
        super(props);
        this.state = { 
            compound: this.props.entry ? this.props.entry.compound : '', 
            quantity: this.props.entry ? this.props.entry.quantity.toString() : '', 
            timestamp: this.props.entry ? moment(this.props.entry.timestamp) : moment(), 
            calendarFocused:false, 
            error:''
        };    
    }

    onCompoundChange = (e) => {
        const compound = e.target.value;
        this.setState(()=>({compound}));
    }
    onQuantityChange = (e) => {
        const quantity = e.target.value;
        if(quantity.match(/^\d*(\.\d{0,3})?$/)){
            this.setState(()=>({quantity}));
        }
    }
    onDateChange = (timestamp) => {        
        if(timestamp){
            this.setState(()=>({timestamp}));
        }
    }
    onFocusChange = ({focused}) => {
        this.setState(()=>({calendarFocused:focused}))
    }
   
    handleOnSubmit = (e) => {
        e.preventDefault();
        const { compound, quantity } = this.state;
        if(!compound || !quantity ){
            this.setState({
                error: 'Please provide compound name and quantity.'
            });
            
        }else{
            this.setState({ error: '' });
            this.props.onFormSubmit({
                compound: this.state.compound,
                quantity: parseFloat(this.state.quantity),
                timestamp: this.state.timestamp.valueOf()
            }); 
        }          
    }
    render(){
        return (
            <React.Fragment>
                {!!this.state.error && <span className="validation-error-msg">{this.state.error}</span>}
                <form  className="form entry-form" onSubmit={this.handleOnSubmit}>
                    <input 
                        type="text" 
                        className="form-input" 
                        name="compound"
                        value={this.state.compound} 
                        onChange={this.onCompoundChange}
                        placeholder="Compound..."
                        autoFocus
                        />
                    <input 
                        type="text" 
                        className="form-input" 
                        name="quantity"
                        value={this.state.quantity} 
                        onChange={this.onQuantityChange}
                        placeholder="Quantity..."
                        />
                    <SingleDatePicker 
                        date={this.state.timestamp}
                        onDateChange = {this.onDateChange}
                        focused = {this.state.calendarFocused}
                        onFocusChange ={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={()=> false}
                        />
                    {
                        this.props.isEdit && 
                        <button 
                            className="btn btn-default btn-delete"
                            onClick={(e)=>{ 
                                e.preventDefault();
                                this.props.onRemoveEntry(this.props.match.params.id);
                            }}
                            >
                            <FontAwesomeIcon icon={faTrashAlt} style={{marginRight:'.8rem'}}/>   
                            Delete
                        </button> 
                    }
                    <div className="btn-block-group">
                        <button className="btn btn-primary btn-md">Submit</button>
                    </div>
                </form>
            </React.Fragment>
        );
    }
}

const EntryFormWithRouter = withRouter(EntryForm)

export default connect()(EntryFormWithRouter);