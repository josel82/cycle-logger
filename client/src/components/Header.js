import React, { Component } from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import { CSSTransition } from 'react-transition-group';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSignOutAlt } from '@fortawesome/free-solid-svg-icons';
import { faArrowLeft } from '@fortawesome/free-solid-svg-icons';

import { startLogout } from '../actions/auth';

class Header extends Component {
    constructor(props){
        super(props);
        this.state = {
            isOpened: false,
            showBackBtn: false
        }

    }
    componentDidUpdate(prevProps) { 
        if (this.props.location !== prevProps.location) {
            
            this.setState({showBackBtn: this.props.location.pathname !== "/dashboard"});
        }
    }
    toggleMenu = () => {
        this.setState({isOpened:!this.state.isOpened});
    }
    navigateBack = () =>{
        this.props.history.push(this.props.path);
    }
    render(){
        return (         
            <header className="header">
                
                <CSSTransition 
                    in={this.state.showBackBtn}
                    appear={true}
                    timeout={600}
                    classNames="fade"
                    onExited={()=>{

                    }}>
                    <span 
                        className={
                            `btn-back ${this.props.location.pathname === '/dashboard' && 'fade-exit-done' }`
                        } 
                        onClick={this.navigateBack}>
                        <FontAwesomeIcon icon={faArrowLeft} />
                    </span>
                </CSSTransition>
                
                <div className="logo-box">
                    <img src="/assets/img/logo-lg-inv.png" alt="Logo" className="logo"/>
                </div>
                <nav className={`header__navigation ${this.state.isOpened ? 'header__navigation--open':''}`}>
                    <ul> 
                        <li>
                            <NavLink to={'/'} onClick={this.props.startLogout}>
                                <FontAwesomeIcon icon={faSignOutAlt} className="header__navigation--icon" />Log out
                            </NavLink>
                        </li>
                    </ul>
                </nav>
                <div className={`menu-toggle ${this.state.isOpened ? 'open':''}`} 
                     onClick={this.toggleMenu}>
                    <div className="hamburger"></div>
                </div>
            </header>
        );
    }
}
 


const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

const HeaderWithRouter = withRouter(Header)

export default connect(undefined, mapDispatchToProps)(HeaderWithRouter);