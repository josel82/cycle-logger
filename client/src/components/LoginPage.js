import React, {Component} from 'react';
import { connect } from 'react-redux';

import { startLogin } from '../actions/auth';

export class LoginPage extends Component {

    handleLogin = () => {
        this.props.onLogin();
    }
    render(){
        return (
            <div className="login-container">
              <h1 className="heading-primary">
                  <span className="heading-primary__parcial__bold">C</span>ycle
                  <br/>
                  <span className="heading-primary__parcial__bold">L</span>ogger
              </h1>
              <div className="btn btn-google btn-md" onClick={this.handleLogin}>
                  <span className="btn-google__logo"></span>
              </div>
            </div>
        );
    }
} 
 
 

const mapDispatchToProps = (dispatch) => ({
    onLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);

