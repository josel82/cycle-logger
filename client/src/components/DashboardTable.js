import React, { Component } from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleDown } from '@fortawesome/free-solid-svg-icons';
import { faPlus } from '@fortawesome/free-solid-svg-icons';

import Entry from './Entry';
import getVisibleEntries from '../selectors/entries';
import { sortByQuantity, sortByDate } from '../actions/filters';



export class DashboardTable extends Component {

    handleSortByQuantity = () => {
        this.props.sortByQuantity();
    }    

    handleSortByDate = () => {
        this.props.sortByDate();
    }

    renderEntries = () => {
        if(this.props.entries.length > 0) {
            return (
                this.props.entries.map((entry, i) => (
                    <Entry 
                        id = {entry.id}
                        compound={entry.compound} 
                        quantity={entry.quantity} 
                        timestamp={entry.timestamp} key={i} 
                        />
                ))
            )
        }else{
            return <tr><td colSpan="3" className="table__cell--spread">No Items</td></tr>
        }
    }

    render(){
        return (
            <React.Fragment>
                <table className="table">
                    <colgroup>
                        <col style={{width:'50%'}}/>
                        <col style={{width:'20%'}}/>
                        <col style={{width:'30%'}}/>
                    </colgroup>  
                    <tbody>
                        <tr>
                            <th>Compound</th>
                            <th 
                                onClick={this.handleSortByQuantity}>
                                Qty {this.props.filters.sortBy === 'quantity' && <FontAwesomeIcon icon={faAngleDown}/>}
                            </th>
                            <th 
                                onClick={this.handleSortByDate}>
                                Date {this.props.filters.sortBy === 'date' && <FontAwesomeIcon icon={faAngleDown}/>}    
                            </th>
                        </tr>
                            {
                                this.renderEntries() 
                            }
                    </tbody>
                </table>
                <div className="btn-add btn-floating" onClick={()=>{this.props.history.push('/dashboard/add')}}>
                    <FontAwesomeIcon icon={faPlus}/>
                </div>
            </React.Fragment>
        );
    }
}

const mapStateToProps = (state) => {
    const visibleEntries = getVisibleEntries(state.entries, state.filters);
    return {
      entries: visibleEntries,
      filters: state.filters
    }
} 

const mapDispatchToProps = (dispatch) =>( {
    sortByQuantity: () => dispatch(sortByQuantity()),
    sortByDate: () => dispatch(sortByDate())
});
export default connect(mapStateToProps, mapDispatchToProps)(DashboardTable);