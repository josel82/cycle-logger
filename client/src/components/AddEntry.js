import React, { Component } from 'react';
import { connect } from 'react-redux';

import { startAddEntry } from '../actions/entries'
import EntryForm from './EntryForm';

export class AddEntry extends Component {

    onSubmit = (entry) =>{                
        this.props.onSubmit(entry);
        this.props.history.push('/dashboard');
    }
    render(){
        return (
            <EntryForm onFormSubmit={this.onSubmit}/>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    onSubmit: (entry) => dispatch(startAddEntry(entry))
});

export default connect(undefined, mapDispatchToProps)(AddEntry);