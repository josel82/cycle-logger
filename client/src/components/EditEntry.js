import React, { Component } from 'react';
import { connect } from 'react-redux';

import { startEditEntry, startRemoveEntry } from '../actions/entries';
import EntryForm from './EntryForm';

export class EditEntry extends Component {

    onEditEntry = (updates) =>{
        const id = this.props.entry.id;                
        this.props.onEditEntry(id, updates);
        this.props.history.push('/dashboard');
    }
    onRemoveEntry = (id) => {
        this.props.onRemoveEntry(id);
        this.props.history.push('/dashboard');
    }
    render(){
        return (
            <EntryForm 
                onFormSubmit={this.onEditEntry} 
                onRemoveEntry={this.onRemoveEntry}
                entry={this.props.entry} 
                isEdit={true} 
                />
        );
    }
}

const mapStateToProps = (state, props) =>({
    entry: state.entries.find((entry)=> (entry.id === props.match.params.id)),
});

const mapDispatchToProps = (dispatch) => ({
        onEditEntry: (id, entry) => dispatch(startEditEntry(id,entry)),
        onRemoveEntry: (id) => dispatch(startRemoveEntry(id))
});


export default connect(mapStateToProps, mapDispatchToProps)(EditEntry);