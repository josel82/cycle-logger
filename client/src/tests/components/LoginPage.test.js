import React from 'react';
import { shallow } from 'enzyme';

import { LoginPage } from '../../components/LoginPage';

test('should render LoginPage correctly', ()=>{
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
}); 

test('should handle onLogin', () => {
    const onLogin = jest.fn();
    const wrapper = shallow(<LoginPage onLogin={onLogin}/>);
    wrapper.find('.btn').simulate('click');
    expect(onLogin).toHaveBeenCalled();
});