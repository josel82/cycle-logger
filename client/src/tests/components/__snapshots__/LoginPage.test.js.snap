// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="login-container"
>
  <h1
    className="heading-primary"
  >
    <span
      className="heading-primary__parcial__bold"
    >
      C
    </span>
    ycle
    <br />
    <span
      className="heading-primary__parcial__bold"
    >
      L
    </span>
    ogger
  </h1>
  <div
    className="btn btn-google btn-md"
    onClick={[Function]}
  >
    <span
      className="btn-google__logo"
    />
  </div>
</div>
`;
